ibm/service/power/resource_ibm_pi_volume_bulk.go

```diff
@@ -0,0 +1,537 @@
+							Computed:    true,
+							Description: "List of replication sites for volume replication.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Type:        schema.TypeList,
+						},
+						Attr_ReplicationType: {
+							Computed:    true,
+							Description: "The replication type of the volume 'metro' or 'global'.",
+							Type:        schema.TypeString,
+						},
+						Attr_VolumeID: {
+							Computed:    true,
+							Description: "The unique identifier of the volume.",
+							Type:        schema.TypeString,
+						},
+						Attr_VolumeStatus: {
+							Computed:    true,
+							Description: "The status of the volume.",
+							Type:        schema.TypeString,
+						},
+						Attr_WWN: {
+							Computed:    true,
+							Description: "The world wide name of the volume.",
+							Type:        schema.TypeString,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+		},
+	}
+}
+
+func ResourceIBMPIVolumeBulkValidator() *validate.ResourceValidator {
+	validateSchema := make([]validate.ValidateSchema, 0)
+	validateSchema = append(validateSchema,
+		validate.ValidateSchema{
+			Identifier:                 "pi_affinity",
+			ValidateFunctionIdentifier: validate.ValidateAllowedStringValue,
+			Type:                       validate.TypeString,
+			Required:                   true,
+			AllowedValues:              "affinity, anti-affinity"})
+	IBMPIVolumeBulkResourceValidator := validate.ResourceValidator{
+		ResourceName: "ibm_pi_volume",
+		Schema:       validateSchema}
+	return &IBMPIVolumeBulkResourceValidator
+}
+
+func resourceIBMPIVolumeBulkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	name := d.Get(Arg_VolumeName).(string)
+	size := int64(d.Get(Arg_VolumeSize).(float64))
+	var shared bool
+	if v, ok := d.GetOk(Arg_VolumeShareable); ok {
+		shared = v.(bool)
+	}
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	body := &models.MultiVolumesCreate{
+		Name:      &name,
+		Shareable: &shared,
+		Size:      &size,
+	}
+	body.Count = int64(d.Get(Arg_Count).(int))
+	if v, ok := d.GetOk(Arg_VolumeType); ok {
+		volType := v.(string)
+		body.DiskType = volType
+	}
+	if v, ok := d.GetOk(Arg_VolumePool); ok {
+		volumePool := v.(string)
+		body.VolumePool = volumePool
+	}
+	if v, ok := d.GetOk(Arg_ReplicationEnabled); ok {
+		replicationEnabled := v.(bool)
+		body.ReplicationEnabled = &replicationEnabled
+	}
+	if v, ok := d.GetOk(Arg_ReplicationSites); ok {
+		if d.Get(Arg_ReplicationEnabled).(bool) {
+			body.ReplicationSites = flex.FlattenSet(v.(*schema.Set))
+		} else {
+			return diag.Errorf("Replication (%s) must be enabled if replication sites are specified.", Arg_ReplicationEnabled)
+		}
+	}
+	if ap, ok := d.GetOk(Arg_AffinityPolicy); ok {
+		policy := ap.(string)
+		body.AffinityPolicy = &policy
+
+		if policy == "affinity" {
```

Comment:
Could you use a constant for `affinity` ?

Reply:
updated